flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-elasticsearch/src/main/java/org/apache/flink/cdc/connectors/elasticsearch/sink/ElasticsearchDataSinkOptions.java

```diff
@@ -0,0 +1,104 @@
+    public static final ConfigOption<String> HOSTS =
+            ConfigOptions.key("hosts")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription(
+                            "The comma-separated list of Elasticsearch hosts to connect to.");
+
+    /** The maximum number of actions to buffer for each bulk request. */
+    public static final ConfigOption<Integer> MAX_BATCH_SIZE =
+            ConfigOptions.key("batch.size.max")
+                    .intType()
+                    .defaultValue(500)
+                    .withDescription(
+                            "The maximum number of actions to buffer for each bulk request.");
+
+    /** The maximum number of concurrent requests that the sink will try to execute. */
+    public static final ConfigOption<Integer> MAX_IN_FLIGHT_REQUESTS =
+            ConfigOptions.key("inflight.requests.max")
+                    .intType()
+                    .defaultValue(5)
+                    .withDescription(
+                            "The maximum number of concurrent requests that the sink will try to execute.");
+
+    /** The maximum number of requests to keep in the in-memory buffer. */
+    public static final ConfigOption<Integer> MAX_BUFFERED_REQUESTS =
+            ConfigOptions.key("buffered.requests.max")
+                    .intType()
+                    .defaultValue(1000)
+                    .withDescription(
+                            "The maximum number of requests to keep in the in-memory buffer.");
+
+    /** The maximum size of batch requests in bytes. */
+    public static final ConfigOption<Long> MAX_BATCH_SIZE_IN_BYTES =
+            ConfigOptions.key("batch.size.max.bytes")
+                    .longType()
+                    .defaultValue(5L * 1024L * 1024L)
+                    .withDescription("The maximum size of batch requests in bytes.");
+
+    /** The maximum time to wait for incomplete batches before flushing. */
+    public static final ConfigOption<Long> MAX_TIME_IN_BUFFER_MS =
+            ConfigOptions.key("buffer.time.max.ms")
+                    .longType()
+                    .defaultValue(5000L)
+                    .withDescription(
+                            "The maximum time to wait for incomplete batches before flushing.");
+
+    /** The maximum size of a single record in bytes. */
+    public static final ConfigOption<Long> MAX_RECORD_SIZE_IN_BYTES =
+            ConfigOptions.key("record.size.max.bytes")
+                    .longType()
+                    .defaultValue(10L * 1024L * 1024L)
+                    .withDescription("The maximum size of a single record in bytes.");
+
+    /** The Elasticsearch index name to write to. */
+    public static final ConfigOption<String> INDEX =
+            ConfigOptions.key("index")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription("The Elasticsearch index name to write to.");
```

Comment:
Actually, I don't see any place that uses this option.
What about setting all index name to this value if this config was set, otherwise use tableId as index name, like [topic in Kafka connector](https://nightlies.apache.org/flink/flink-cdc-docs-release-3.1/docs/connectors/pipeline-connectors/kafka/#pipeline-connector-options).